### PR TITLE
Feature/post conversation members

### PR DESCRIPTION
**Pull request by KaterinaWire**
## What's new in this PR?

- update tests for POST conversations/
- add the endpoint PUT conversations/:cid/members/:uid
- add test for the PUT conversations/:cid/members/:uid

### CODE REVIEW

**Comment by zenkins** [comment]
Can one of the admins verify this patch?

File: Source/Conversations/MockTransportSession+conversations.m [hunk 1]
```diff
@@ -217,6 +221,22 @@ - (ZMTransportResponse *)processPutConversation:(NSString *)conversationId paylo
...
+{
+    MockConversation *conversation = [self conversationByIdentifier:conversationId];
+    if (conversation == nil) {
+        return [ZMTransportResponse responseWithPayload:nil HTTPStatus:404 transportSessionError:nil];
+    }
+    
+    NSString *conversationRole = [payload optionalStringForKey:@"conversation_role"];
+    MockUser *user = [self fetchUserWithIdentifier:memberId];
+    if (conversationRole != nil && user!= nil) {
+        user.role = conversationRole;
```
Comment (marcoconti83) [inline Source/Conversations/MockTransportSession+conversations.m]: I would add a check: is the user part of this conversation? If not, return an error (`400` or something similar)

Comment (KaterinaWire) [reply Source/Conversations/MockTransportSession+conversations.m]: Fixed

File: Source/Conversations/MockTransportSession+conversations.m [hunk 2]
```diff
@@ -229,12 +229,25 @@ - (ZMTransportResponse *)processPutMembersInConversation:(NSString *)conversatio
...
+    if (conversationRole == nil) {
+        return [ZMTransportResponse responseWithPayload:@{@"error":@"no conversation_role in payload"} HTTPStatus:400 transportSessionError:nil];
+    }
+    
     MockUser *user = [self fetchUserWithIdentifier:memberId];
-    if (conversationRole != nil && user!= nil) {
-        user.role = conversationRole;
+    if(user == nil) {
+        return [ZMTransportResponse responseWithPayload:@{
+                                                          @"code" : @403,
```
Comment (marcoconti83) [inline Source/Conversations/MockTransportSession+conversations.m]: Make it `400 bad request` instead. `403` is for authentication errors.

**Pull request merged**